### PR TITLE
fixed advanced all bug, by inserting proper status to database

### PR DESCRIPTION
**Pull request by iknight7000**
Co-authored-by: Chris <chris-chilton@hotmail.com>

# Pull Request Template

## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

This bug fix ensures that the advance all API call, progresses the conversation by one step and updates the conversations table accordingly. 
video [link](https://www.loom.com/share/0588e222cf954bb29e50989c902bfff2)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

This bug was tested by progressing through a conversation on a local set-up

- [x ] Test A
- [ ] Test B

**Test Configuration**:
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

## Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: app/db.py [hunk 1]
```diff
@@ -449,7 +449,7 @@ def insert_data_conversations(self, data):
                 last_value = 0
             if data.id is None:
                 last_value += 1
-            data.id = last_value
```
Comment (iknight7000) [inline app/db.py:452]: Moving this element into the if statement was the big fix.

File: app/db.py [hunk 2]
```diff
@@ -449,7 +449,7 @@ def insert_data_conversations(self, data):
...
+                data.id = last_value
             session.add(data)
```
Comment (ChristopherKchilton) [inline app/db.py:453]: This was the error. The lack of indentation tried to update everything as the last object in the database.

**Review by ChristopherKchilton — APPROVED** [review]
I approve of these changes. This is just a minor fix that solved the advance function key error due to it trying to assign everything as the last id number which is suppose to be unique.

File: app/db.py (hunk 1 above)
Comment (gaben3722) [inline app/db.py:452]: A simple fix but something that had to be done. Approved.

**Review by gaben3722 — APPROVED** [review]

**Review by mikecarrier4 — APPROVED** [review]
good catch, small error quickly fixed

**Pull request merged**